### ISSUE TITLE
'Mock dependencies using Mockito' page issue

### ISSUE THREAD

**Issue by i3games** [flutter/website#5488]
Page URL: https://flutter.dev/docs/cookbook/testing/unit/mocking.html
Page source: https://github.com/flutter/website/tree/master/src/docs/cookbook/testing/unit/mocking.md

Description of issue:

This page neeeds to be updated. 

(1) references a previous version of the https://flutter.dev/docs/cookbook/networking/fetch-data recipe. The current recipe fetches `Album`not `Post`

(2) does not contain workable code. It introduces a code snippet where `http.Client`provided as a parameter to the `fetchPost` function. This breaks the old code and it should show how and where the client is created.

(3) uses an outdated `http` package with substantial brealking changes to the currrent version.

The recipe needs a complete overhaul adressing (1) to (3).

**Comment by sfshaza2** [comment]
@miquelbeltran, would you mind adding this to your list?

**Comment by miquelbeltran** [comment]
Yep, the mockito codelab will need some work indeed, because Mockito works differently as it is currently with null safety and requires to do code generation.

**Comment by miquelbeltran** [comment]
Hey @i3games I have submitted a PR addressing your points, you can take a look at #5492 

Regarding point 2:

> This breaks the old code and the recipe does not show how and where the http client is created.

I think that this is out of the scope of the cookbook recipe, as the goal is to explain how mocks work rather how the `http` package works. In this case developers would have to use `http.Client()` to create a Client instance. But if it not obvious and @sfshaza2 agrees I can add a short explanation for that in the cookbook recipe.

**Comment by sfshaza2** [comment]
I agree that this recipe is not about how to use the http package, but I also don't object to adding a single sentence or two.

**Comment by miquelbeltran** [comment]
thanks! I will update the PR then

**Comment by i3games** [comment]
I would recommend to add a full working example at the end as it is common in other Flutter recipes. The goal of a recipe should be self-contained and working code that I can take without having to understand how to build contextual dependencies.

**Comment by sfshaza2** [comment]
This makes a lot of sense to me. What do you think, @miquelbeltran?

**Comment by miquelbeltran** [comment]
I've added now the complete example, copying over the parts from the fetch-data recipe but modified to be testable

**Comment by miquelbeltran** [comment]
this can be closed now! :)
